numpy/_core/src/umath/ufunc_object.c

```diff
@@ -4486,6 +4486,15 @@ ufunc_generic_fastcall(PyUFuncObject *ufunc,
         return override;
     }
 
+    /* Warn if "where" is used without "out", issue 29561 */
+    if ((where_obj != NULL) && (full_args.out == NULL) && (out_obj != Py_None)) {
```

Comment:
```suggestion
    if ((where_obj != NULL) && (full_args.out == NULL) && (out_obj == NULL)) {
```
Seems a bit clearer to me?

Reply:
It seems clearer to me to compare to the desired `Py_None` than to rely on logic

Reply:
I suppose if you see it this way, but actually we should make it:
```
!out_is_passed_by_position && out_obj == NULL
```
that makes it clear: out was not passed explicitly.  But more importantly, there is a concrete reason to avoid `!= None` and that is:
```
np.frexp(4., out=(None, None), where=True)
```

Reply:
OK, let's go with your suggestion

Reply:
Also need to avoid a case where `out` is an actual array!